### ISSUE TITLE
primary db crash

### ISSUE THREAD

**Issue by weickdev** [arangodb/arangodb#3655]
## my environment running ArangoDB
I'm using the latest ArangoDB of the respective release series:
- [ ] 2.8
- [ ] 3.0
- [ ] 3.1
- [x] 3.2
- [ ] self-compiled devel branch

Mode:
- [x] Cluster
- [ ] Single-Server

Storage-Engine:
- [ ] mmfiles
- [x] rocksdb

On this operating system:
- [ ] DCOS on
  - [ ] AWS
  - [ ] Azure
  - [ ] own infrastructure
- [ ] Linux 
  - [ ] Debian .deb
  - [ ] Ubuntu .deb
  - [ ] SUSE   .rpm
  - [x] RedHat .rpm
  - [ ] Fedora .rpm
  - [ ] Gentoo
  - [ ] docker - official docker library
  - [ ] other:
- [ ] Windows, version: 
- [ ] MacOS, version:

My primay db process crashes many times when I insert massive documents. I find below warning message:
```
2017-11-10T22:46:55 [7840] INFO {replication} connected to master at tcp://192.168.1.62:18558, id 36969257467309, version 3.2, last log tick 147279601
2017-11-10T22:46:55 [7840] INFO {replication} connected to master at tcp://192.168.1.62:18558, id 36969257467309, version 3.2, last log tick 147279601
2017-11-10T22:46:55 [7840] INFO {replication} fetching master log for collection 's8010284' from tick 147279601, barrier: 0
2017-11-10T22:47:10 [7840] INFO {replication} connected to master at tcp://192.168.1.62:18558, id 36969257467309, version 3.2, last log tick 147279865
2017-11-10T22:47:10 [7840] INFO {replication} connected to master at tcp://192.168.1.62:18558, id 36969257467309, version 3.2, last log tick 147279865
2017-11-10T22:47:10 [7840] INFO {replication} fetching master log for collection 's8010290' from tick 147279865, barrier: 0
2017-11-10T23:20:30 [7840] WARNING caught exception in rocksdb background thread: std::bad_alloc
2017-11-10T23:20:32 [9082] TRACE {startup} Action::prepare
2017-11-10T23:20:32 [9082] TRACE {startup} FileDescriptors::prepare
2017-11-10T23:20:32 [9082] TRACE {startup} Greetings::prepare
2017-11-10T23:20:32 [9082] INFO ArangoDB 3.2.5 [linux] 64bit, using jemalloc, VPack 0.1.30, RocksDB 5.6.0, ICU 58.1, V8 5.7.0.0, OpenSSL 1.0.1e-fips 11 Feb 2013
2017-11-10T23:20:32 [9082] TRACE {startup} LoggerBuffer::prepare
```
There is another error:
```
2017-11-10T23:22:24 [4403] INFO {replication} fetching master log for collection 's8010269' from tick 152969431, barrier: 0
2017-11-10T23:35:15 [4403] ERROR {memory} process is near the maximum number of memory mappings. current: 319007, maximum: 320000. it may be sensible to increase the maximum number
of mappings per process
2017-11-10T23:35:32 [4403] FATAL out of memory in V8 (Committing semi space failed.)
2017-11-10T23:35:33 [5516] TRACE {startup} Action::prepare
2017-11-10T23:35:33 [5516] TRACE {startup} FileDescriptors::prepare
```
Is 320000 not big enough?

**Comment by dothebart** [comment]
Hi, 
we have fixed some issues with memory management and jemalloc, plus upgraded to a bugfixed V8 version with also memory management issues with ArangoDB 3.2.8.
 
Can you please upgrade and retry whither the issue persists?

**Comment by dothebart** [comment]
Hi, 
Any news on this? Did the upgrade improve the situation for you? can we close this issue?

**Comment by ropeladder** [comment]
Hi dothebart,

I'm having a similar issue using RocksDB in 3.2.8, not sure if it's the exact same.
Linux Mint 18 (Ubuntu), 16gb ram.

This will crash it reliably:
```
FOR n IN 1..10000000
    INSERT {n, otherData: "test test test"} INTO TestNums
    
FOR n IN TestNums
    UPDATE n WITH {otherData: "test some more"} IN TestNums
```

This is the only error I get:
```
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
```

**Comment by dothebart** [comment]
@ropeladder do you also use a cluster? if not, it would be better to open a new issue for that.

**Comment by ropeladder** [comment]
No, I'm just using a single instance. I'll open a new issue.

**Comment by dothebart** [comment]
Ok, will close this as solved with 3.2.8

**Comment by dothebart** [comment]
Ok, 
I think we can close this issue then.
